### PR TITLE
Implement uploading <project_uuid>.hazmapper to tapis.

### PR DESCRIPTION
**Pull request by duckonomy**
## Overview: ##
Implement saving the project file to tapis.

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-1929](https://jira.tacc.utexas.edu/browse/DES-1929)

## Summary of Changes: ##
Adds necessary route/service for the upload process and a new tapis interaction (create file and delete) in `agave.py`
Also added `system_path`, `system_id`, `system_name` to the project model for creating a link to designsafe projects

## Testing Steps: ##
1. Test with the HazMapper GUI (TACC-Cloud/hazmapper#50)

## Notes: ##
I wasn't sure how to test the Tapis part. I've tried creating a mock like the `agave_utils_with_geojson_file_mock` but wasn't sure if that could update on making the call.

### CODE REVIEW

**Comment by reptillicus** [comment]
It looks like there should also be a migration in here too? The additional fields in the project model  will need one.

**Review by nathanfranklin — APPROVED** [review]
LGTM 👍 

Just note two follow on things:
* comment about logging
* potentially deriving 'system_name' on the front end instead of having in model

File: geoapi/routes/public_projects.py [hunk 1]
```diff
@@ -59,6 +60,11 @@ class PublicProjectPointCloudsResource(ProjectPointCloudsResource, metaclass=Hid
     pass
 
 
+@api.route('/<int:projectId>/export/')
```
Comment (nathanfranklin) [inline geoapi/routes/public_projects.py:63]: I don't think we need this as we don't want to offer this route to unauthenticated users, right?

**Review by nathanfranklin — CHANGES_REQUESTED** [review]
Just left one comment to consider and also for 3038e1d74ecc7ed722e3d232b311651b726a81fc, is there a migration that goes along with it (as we have an early commit of this branch deployed on staging). 👍

**Review by nathanfranklin — APPROVED** [review]
Awesome 💯

**Pull request merged**